File: day7/p1.rs
use std::fs;

fn main() {
    let mut input = fs::read_to_string("input.txt")
        .expect("Unable to read file")
        .trim()
        .split(",")
        .map(|x| x.parse::<i32>().unwrap())
        .collect::<Vec<i32>>();
    input.sort();
    let totalelems = input.len();
    let median = input[totalelems / 2];
    let ans = input.into_iter().fold(0, |a, b| (b - median).abs() + a);
    println!("{}", ans);
}

File: 2024/day3/p1.py
import re
[
f := open("input.txt", "r"),
s := f.read(),
print(sum(map(lambda x: (b:=list(map(int, x[4:-1].split(','))))[0]*b[1], (re.findall(r"mul\([0-9]{1,3},[0-9]{1,3}\)", s)))))
]
